fferr/errors.go

```diff
@@ -76,6 +76,7 @@ type Error interface {
 	GRPCStatus() *status.Status
 	ToErr() error
 	AddDetail(key, value string)
+	AddDetails(keysAndValues ...interface{})
```

Comment:
Do we really want to allow values of any type here? Why not limit it to strings to keep things simple?

Reply:
It's just the interface method to do it. I modeled this after how logger.Infow works. The validation is done at the impl level